### DISCUSSION TITLE
I would like to retrieve the names of all the meshes in the scene. How can I achieve this?

### DISCUSSION THREAD

**Discussion by WdsPjm** [vsg-dev/VulkanSceneGraph#865] [Q&A] [answered]
I would like to retrieve the names of all the meshes in the scene. How can I achieve this?

**Comment by robertosfield** [comment, marked as answer]
I move moved this question from vsgExamples Issues to the main VulkaSceneGraph Discussion as Q&A thread as this is best place for such community discussions,

The way that the VSG stores user defined values like names is via the meta-data object system supported in all classes subclassed from vsg::Object, the vsgTutorial has section on meta-data:

https://vsg-dev.github.io/vsgTutorial/foundations/MetaData

The loader will need to set the names, and the vsgXchange::assimp loader does this, so if you have loaded data via it then you should be able to get the names from it.  To help figure things out it can be useful to convert the file to .vsgt ascii format and then have a search through with an editor to see how the meta data is assigned,

**Comment by WdsPjm** [comment]
Thank you very much, I have completed it.
